internal/graphicsdriver/metal/ca/ca_darwin.go

```diff
@@ -106,94 +114,91 @@ func (ml MetalLayer) SetPixelFormat(pf mtl.PixelFormat) {
 //
 // Reference: https://developer.apple.com/documentation/quartzcore/cametallayer/2938720-maximumdrawablecount.
 func (ml MetalLayer) SetMaximumDrawableCount(count int) {
-	e := C.MetalLayer_SetMaximumDrawableCount(ml.metalLayer, C.uint_t(count))
-	if e != nil {
-		panic(errors.New(C.GoString(e)))
+	if count < 2 || count > 3 {
+		panic(errors.New(fmt.Sprintf("failed trying to set maximumDrawableCount to %d outside of the valid range of [2, 3]", count)))
 	}
+	ml.metalLayer.Send(objc.RegisterName("setMaximumDrawableCount:\x00"), count)
 }
 
 // SetDisplaySyncEnabled controls whether the Metal layer and its drawables
 // are synchronized with the display's refresh rate.
 //
 // Reference: https://developer.apple.com/documentation/quartzcore/cametallayer/2887087-displaysyncenabled.
 func (ml MetalLayer) SetDisplaySyncEnabled(enabled bool) {
-	switch enabled {
-	case true:
-		C.MetalLayer_SetDisplaySyncEnabled(ml.metalLayer, 1)
-	case false:
-		C.MetalLayer_SetDisplaySyncEnabled(ml.metalLayer, 0)
+	if runtime.GOOS == "ios" {
```

Comment:
This would not work with Go 1.15 (`ios` is not one of GOOS yet at that time). Why did you change the logic here by the way?

Reply:
The logic should be the same. There was a check in the C code to only run it on IOS. I replaced all runtime.GOOS == "iOS" with just a check if the Object responds to a certain function.

Reply:
The latest change (checking the existence of the method) seems fine.